apps/designer/app/canvas/shared/use-shortcuts.ts

```diff
@@ -1,38 +1,59 @@
 import { useHotkeys } from "react-hotkeys-hook";
 import store from "immerhin";
-import {
-  type Instance,
-  publish,
-  useSubscribe,
-  components,
-} from "@webstudio-is/react-sdk";
+import { type Instance, components } from "@webstudio-is/react-sdk";
 import { shortcuts, options } from "~/shared/shortcuts";
+import { publish, useSubscribe } from "~/shared/pubsub";
 import { useSelectedInstance } from "./nano-states";
 import { copy, paste } from "./copy-paste";
 import {
   useRootInstance,
   useTextEditingInstanceId,
 } from "~/shared/nano-states";
+import { type SelectedInstanceData } from "~/shared/canvas-components";
+
+declare module "~/shared/pubsub" {
+  export interface PubsubMap {
+    cancelCurrentDrag: undefined;
+    deleteInstance: {
+      id: Instance["id"];
+    };
+    openBreakpointsMenu: undefined;
+    selectBreakpointFromShortcut: number;
+    selectInstance?: SelectedInstanceData;
+    togglePreviewMode: undefined;
+    zoom: "zoomOut" | "zoomIn";
+  }
+}
 
 const inputTags = ["INPUT", "SELECT", "TEXTAREA"] as const;
 
 type HandlerEvent = {
-  key: string;
+  key?: string;
   preventDefault?: () => void;
 };
 
 const togglePreviewMode = () => {
-  publish<"togglePreviewMode">({ type: "togglePreviewMode" });
+  publish({ type: "togglePreviewMode" });
 };
 
 const publishSelectBreakpoint = ({ key }: HandlerEvent) => {
+  if (!key) {
+    throw new Error(
+      "`publishSelectBreakpoint` doesn't account for being called without a `key`"
+    );
+  }
   publish({
     type: "selectBreakpointFromShortcut",
-    payload: key,
+    payload: parseInt(key, 10),
```

Comment:
As far as I can tell this is a fix because the subscriber expects to receive an `index`, and that should be a number:
https://github.com/webstudio-is/webstudio-designer/blob/b913ac73d726ad41ca33bda2d44f1bd6fdf09ea9/apps/designer/app/designer/features/breakpoints/use-subscribe-shortcuts.ts#L10

Reply:
nah, lets make the key string there

Reply:
`selectBreakpointFromShortcut` doesn't have to rely on literal keys coming from the events - it's already a "translated" event (sort-of). IMHO when possible it's better to refine data asap, closer to the publisher than to the subscriber. As this is going to be used as an index, I think that it makes sense to convert this here and have that consuming logic slightly simpler.